unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BIWMainController.cs

```diff
@@ -212,13 +218,13 @@ public override void Dispose()
             HUDController.i.builderInWorldMainHud.OnLogoutAction -= ExitEditMode;
         }
 
-        BuilderInWorldTeleportAndEdit.OnTeleportEnd -= OnPlayerTeleportedToEditScene;
+        BIWTeleportAndEdit.OnTeleportEnd -= OnPlayerTeleportedToEditScene;
 
         if (initialLoadingController != null)
             initialLoadingController.Dispose();
 
 
-        BuilderInWorldNFTController.i.OnNFTUsageChange -= OnNFTUsageChange;
+        BIWNFTController.i.OnNFTUsageChange -= OnNFTUsageChange;
```

Comment:
```suggestion
        BIWNFTController.i.OnNFTUsageChange -= OnNFTUsageChange;
```
This singleton irks me badly. We can't replace it by dependency injection?

Reply:
Yes we can, this is an old implementantion ( When we were using the singleton approach) However, I think this is out of this scope ( remove monobehaviours), should we create a tech debt card?